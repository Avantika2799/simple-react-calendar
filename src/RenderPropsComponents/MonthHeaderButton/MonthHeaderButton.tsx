import React, { FC, ReactElement } from 'react'
import classnames from 'classnames'

type Props = {
  children?: ReactElement
  blockClassName?: string
  isEnabled?: boolean
  onClick: () => void
  title?: string
  type: 'prev' | 'next'
  offset?: number
}

const displayName = 'MonthHeaderButton'

const HeaderButton: FC<Props> = ({
  children,
  blockClassName,
  isEnabled,
  type,
  title,
  onClick,
  offset = type === 'next' ? 1 : -1
}) => (
  <button
    className={classnames(`${blockClassName}-${displayName}`, `is-${type}`, {
      'is-disabled': !isEnabled
    })}
    type='button'
    disabled={!isEnabled}
    title={title}
    onClick={onClick}
    data-simple-react-calendar-button={offset}
  >
    {children}
  </button>
)

HeaderButton.displayName = displayName

export default HeaderButton
