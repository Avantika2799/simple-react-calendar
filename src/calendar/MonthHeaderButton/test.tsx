import { cleanup, fireEvent, render, RenderResult } from '@testing-library/react'
import React from 'react'

import MonthHeaderButton, { Props } from './MonthHeaderButton'

afterEach(cleanup)

describe('MonthHeaderButton', () => {
  let api: RenderResult, props: Props, mockedCustomRender: any

  beforeEach(() => {
    props = {
      blockClassName: 'test-class-name',
      enabled: true,
      handleOnClick: jest.fn(),
      type: 'next',
    }
  })

  describe('action handlers', () => {
    beforeEach(() => {
      api = render(<MonthHeaderButton {...props} />)
    })

    test('invoke `#handleOnClick`', () => {
      const { getByTitle } = api

      fireEvent.click(getByTitle(/Next Month/i))

      expect(props.handleOnClick).toHaveBeenCalledTimes(1)
    })
  })

  describe('when type is `next`', () => {
    beforeEach(() => {
      api = render(<MonthHeaderButton {...props} />)
    })

    test('default render', () => {
      const { container } = api

      expect(container).toMatchSnapshot()
    })
  })

  describe('when type is `prev`', () => {
    beforeEach(() => {
      props = {
        ...props,
        type: 'prev',
      }
      api = render(<MonthHeaderButton {...props} />)
    })

    test('default render', () => {
      const { container } = api

      expect(container).toMatchSnapshot()
    })
  })

  describe('when render prop has been passed', () => {
    beforeEach(() => {
      mockedCustomRender = jest.fn(() => null)

      props = {
        ...props,
        customRender: mockedCustomRender,
      }
      api = render(<MonthHeaderButton {...props} />)
    })

    test('render prop has been called properly', () => {
      expect(props.customRender).toHaveBeenCalledTimes(1)
      expect(props.customRender).toHaveBeenCalledWith(props)
    })
  })
})
