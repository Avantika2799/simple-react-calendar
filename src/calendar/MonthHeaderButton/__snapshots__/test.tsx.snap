// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthHeaderButton when type is \`next\` default render 1`] = `
<div>
  <button
    class="test-class-name-header_button is-next"
    data-calendar-value="1"
    title="Next Month"
    type="button"
  >
    &lt;
  </button>
</div>
`;

exports[`MonthHeaderButton when type is \`prev\` default render 1`] = `
<div>
  <button
    class="test-class-name-header_button is-prev"
    data-calendar-value="-1"
    title="Previous Month"
    type="button"
  >
    &gt;
  </button>
</div>
`;
